tests/integration/pkg/client/client.go

```diff
@@ -49,10 +52,30 @@ func NewK8sClient() (*K8sClient, error) {
 	}
 
 	h := &K8sClient{kubeClient: kubeClient, controllerClient: controllerClient}
+
+	h.cli, err = findCLI()
+	if err != nil {
+		fmt.Println("failed to find oc or kubectl cli")
+		os.Exit(1)
+	}
 	return h, nil
 }
 
 // Kube returns the clientset for Kubernetes upstream.
 func (c *K8sClient) Kube() kubernetes.Interface {
 	return c.kubeClient
 }
+
+// findCLI returns the first found CLI compatible with oc/kubectl
+func findCLI() (string, error) {
+	for _, cli := range []string{"oc", "kubectl"} {
+		cmd := exec.Command(cli, "version")
```

Comment:
`oc version` or `kubectl version` might take time to return IMO, since they are checking for the server version by default. Perhaps it would make more sense to run `version --client` instead?
Actually, I was wondering if we need to execute them (maybe to check later the version returned?). Just checking whether either of these programs is present and resolvable (using [`exec.LookPath`](https://pkg.go.dev/os/exec#LookPath)) could probably be sufficient? WDYT?

Reply:
Actually, I can see that the `Makefile` already contains a similar logic based on a `K8S_CLI` environment variable: https://github.com/feloy/registry-operator/blob/feature-1239/test-with-kubectl/Makefile#L33-L44
Could we rely on that environment variable here as well?
I was checking this, and noticed that running `make install-cert` used `kubectl`, while the tests used `oc`. It would make more sense to me to rely on the same binary.